contracts/DomainRegistryV2.sol

```diff
@@ -101,27 +170,28 @@ contract DomainRegistryV2 is Initializable, OwnableUpgradeable {
         if (parentDomainOwner == address(0)) {
             parentReward = 0;
         } else {
-            parentReward = price * REWARD_PERCENT_OWNER / 100;
+            parentReward = (price * REWARD_PERCENT_OWNER) / 100;
         }
         uint256 ownerReward = price - parentReward;
 
         domains[rootDomain] = msg.sender;
         shares[msg.sender] += ownerReward;
-        payable(parentDomainOwner).transfer(parentReward);
+        emit DomainRegistered(msg.sender, rootDomain);
 
+        payable(parentDomainOwner).transfer(parentReward);
         uint256 excess = msg.value - price;
         if (excess > 0) {
             payable(msg.sender).transfer(excess);
         }
```

Comment:
It's better not to send the excess at this point. You can simplify the logic if just check **the strict equality** of msg.sender to the price in the relevant modifier. Otherwise you complicate the code with no strong reason. However, let me know if I miss something.

Apart from that, if the msg.sender is a contract, you are leaving it without a flexibility to process the ether in some `receive()` or `fallback() `function as transfer allow for only 2300 gas.
 -1 point

Reply:
fixed, updated `checkSufficientEtn` modifier which check exact amount. Previously I sent 1 etn, although the price could be 0.75 etc.